src/satellite_images_nso_tif_model_iterator/tif_model_iterator/tif_model_iterator.py

```diff
@@ -149,7 +149,7 @@ def _create_and_write_part_output(
             not self.__check_if_part_exists(
                 self.output_file_name_generator.generate_part_output_path(x_step)
             )
-            and self.skip_done_part is True
+            and self.skip_done_part is not True
```

Comment:
This if/else does not correctly approach existing parts as desired. Change skip_done_part into different naming.

Reply:
Rewrite to "always_all_parts" instead of "skip_done_part". Then logic is if (always_all_parts | not file exists):/